core/src/main/java/com/linecorp/armeria/internal/common/util/TemporaryThreadLocals.java

```diff
@@ -106,10 +120,19 @@ void clear() {
         return allocateByteArray(minCapacity);
     }
 
+    /**
+     * Release the byte array to be reused.
+     */
+    public void releaseByteArray() {
+        byteArrayLock = false;
+    }
+
     private byte[] allocateByteArray(int minCapacity) {
         final byte[] byteArray = new byte[minCapacity];
         if (minCapacity <= MAX_BYTE_ARRAY_CAPACITY) {
             this.byteArray = byteArray;
+        } else {
+            byteArrayLock = false;
```

Comment:
> hidden from outside.

But there is no need to release the lock as returned byte array is not stored and will not be reused. Or did you mean something different?

> guide explicitly

By the way, as you said, it would be better to guide the flow in not only method-level Javadoc, but also class-level Javadoc. I will update.

Reply:
> But there is no need to release the lock as returned byte array is not stored and will not be reused.

Yes, techically that is right. However, don't we need to make sure to release the values created from `TemporaryThreadLocals.get().xxx()` whether they are reused or not.

Reply:
I think we don't need to care because `TemporaryThreadLocals.get().xxx()` allocates, but does not manage anymore. In other words, when requested capacity is over `MAX_*_ARRAY_CAPACITY`, the caller is in charge of the responsibility of the resource as a reference only remains in the caller. If the other caller get an exception by the unmanaged short-living object, it seems weird. For above case, we can technically say that there are two locks, explicit one in `TemporaryThreadLocals` and implicit one in the caller.

Of course, to reduce the confusion, guiding to call `release*` for all cases(managed and unmanaged) is okay.